content/posts/2025-01-30-ebpf-flows-filtering.md

```diff
@@ -0,0 +1,228 @@
+The Following Diagram shows how eBPF filtering is done
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-flows-filtering.png')}" alt="eBPF Flows Filtering Processing">
+
+Note:
+- The maximum number of eBPF flow filter rules is limited to 16 to ensure conservative memory
+usage when this feature is enabled.
+
+- rules with duplicate CIDRs are rejected via validation webhook.
+
+- Both IPv4 and IPv6 formats are supported.
+
+- In case users wanted to match on any CIDR, default Null CIDR `0.0.0.0/0` can be used.
+
+## Key Use Cases
+
+🚀 Reducing Observability Overhead – Avoid logging irrelevant flows in high-traffic Kubernetes clusters.
+
+🔐 Security Filtering – Focus on anomalous or suspicious traffic while ignoring normal flows.
+
+🌐 Network Performance Monitoring – Capture only high-latency or dropped-packet flows for troubleshooting.
+
+### Filter EastWest and NorthSouth flows
+
+Let's allow serviceIP 172.30.100.64:80 with sample 10
+and allow pods traffic between subnet 10.128.0.0/16 and 10.129.0.0/16 with sampling 20 and reject everything else
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          sampling: 10
+        - action: Accept
+          cidr: 10.128.0.0/16
+          peerCIDR: 10.129.0.0/16
+          sampling: 20
+        - action: Reject
+          cidr: 0.0.0.0/0
+``` 
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-svc-and-pods-flows.png')}" alt="eBPF Flows Filtering Kubernetes NorthSouth and EastWest Flows">
+
+### Filter flows with packet drops
+
+let's filter any kubernetes service flows with a packet drop and reject everything else,
+please note for this use case to work need to enable `PacketDrop` feature which requires `privileged` flag
+to be set to `true` as shows in the following sample configuration
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      privileged: true
+      features:
+      - PacketDrop
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          pktDrops: true
+        - action: Reject
+          cidr: 0.0.0.0/0
+```
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-filter-svc-pkt-drops.png')}" alt="eBPF Flows Filtering Kubernetes Services with Packet Drop">
+
+### Filter TCP flows with TCP Flags detect TCP SYN Flood attack
+
+Using TCP Flags filtering can help detect when your cluster is under TCP Syn Flood attack
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 0.0.0.0/0
+          protocol: TCP
+          tcpFlags: "SYN"
+          sampling: 1
+```
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-filter-with-tcpflags.png')}" alt="eBPF Flows Filtering TCP flows using TCP flags">
+
+### Filter DNS query over ports 53 and 5353 for both TCP and UDP
+
```

Comment:
This does not have a title, like the other examples

Reply:
added one